requests_oauthlib/oauth2_session.py

```diff
@@ -334,13 +349,42 @@ def request(self, method, url, data=None, headers=None, withhold_token=False,
                     log.debug('Auto refresh is set, attempting to refresh at %s.',
                               self.auto_refresh_url)
 
-                    # We mustn't pass auth twice.
-                    auth = kwargs.pop('auth', None)
-                    if client_id and client_secret and (auth is None):
+                    # Someday this code can be eliminated, we should be able to
+                    # simply pass **self.auto_refresh_kwargs to
+                    # `refresh_token()` what follows is to maintain
+                    # compatibility
+
+                    # Start with kwargs explicitly requested for refresh.
+                    refresh_kwargs = self.auto_refresh_kwargs.copy()
```

Comment:
I wonder, isn't it going to change the semantic of `auto_refresh_kwargs` parameter? Previously it was intended to contain only key-value pairs to include into the body of the request, but now it also holds arguments of `refresh_token()` method such as `auth`, `timeout`, `proxies`, etc. If somebody wants to send `auth` parameter in the request body together with `refresh_token` and `grant_type`, this change will break their code.